src/zkevm_specs/evm_circuit/execution/return_revert.py

```diff
@@ -24,36 +25,42 @@ def return_revert(instruction: Instruction):
 
     rwc_delta = 3
 
+    callee_gas_left = instruction.curr.gas_left
     if instruction.curr.is_create and is_success:
         # A. Returns the specified memory chunk as deployment code.
 
-        # TODO: Untested case.  Test it once create Tx is implemented, and once
-        # CREATE/CREATE2 are implemented.
         callee_address = instruction.call_context_lookup(CallContextFieldTag.CalleeAddress)
-        reversion_info = instruction.reversion_info()
         code_hash, code_hash_prev = instruction.account_write(
             callee_address, AccountFieldTag.CodeHash
         )
         instruction.constrain_equal(code_hash_prev, RLC(EMPTY_HASH))
         instruction.constrain_equal(code_hash, instruction.curr.aux_data)
 
+        # verify bytecode size less than 24,576 bytes
+        instruction.range_lookup(return_length, MAX_CODE_SIZE)
+
+        # gas cost of CREATE = GAS_COST_CREATE + memory expansion + GAS_COST_CODE_DEPOSIT * len(byte_code)
+        # first two part were handled in create.py
+        callee_gas_left = callee_gas_left - return_length * GAS_COST_CODE_DEPOSIT
+
         # Return a memory chunk as deployment code by copying each byte from
         # callee's memory to bytecode, using the copy circuit.
         copy_length = return_length
-        copy_rwc_inc, _ = instruction.copy_lookup(
-            instruction.curr.call_id,  # src_id
-            CopyDataTypeTag.Memory,  # src_type
-            code_hash,  # dst_id
-            CopyDataTypeTag.Bytecode,  # dst_type
-            return_offset,  # src_addr
-            return_end,  # src_addr_boundary
-            FQ(0),  # dst_addr
-            copy_length,  # length
-            instruction.curr.rw_counter + instruction.rw_counter_offset,
-        )
-        instruction.constrain_equal(copy_rwc_inc, copy_length)  # rwc += copy_length
-        instruction.rw_counter_offset += int(copy_rwc_inc)
-        rwc_delta += int(copy_length)
+        if int(return_length) > 0:
+            copy_rwc_inc, _ = instruction.copy_lookup(
+                instruction.curr.call_id,  # src_id
+                CopyDataTypeTag.Memory,  # src_type
+                code_hash.expr(),  # dst_id
+                CopyDataTypeTag.Bytecode,  # dst_type
+                return_offset,  # src_addr
+                return_end,  # src_addr_boundary
+                FQ(1),  # dst_addr
```

Comment:
typo?
```suggestion
                FQ(0),  # dst_addr
```

Reply:
my bad, fixed in [e5acf67](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/355/commits/e5acf67005cc8571ca85bdd0006b517e39ac0cb3)